config/experiments.go

```diff
@@ -11,4 +11,5 @@ type Experiments struct {
 	AcceleratedDHTClient          experimentalAcceleratedDHTClient `json:",omitempty"`
 	OptimisticProvide             bool
 	OptimisticProvideJobsPoolSize int
+	GatewayOverLibp2p             bool
```

Comment:
Let's switch this to a `Flag`  (or at least at add omitempty) so that if we change this from a bool to a struct later or remove it from being Experimental that users who haven't opted in aren't impacted.

Reply:
Went with a bool and omitempty here so that it doesn't show up in the experiments by default which means if we make changes here (e.g. move it from Experiments, change it to a struct, etc.) people who haven't opted in aren't impacted.

If it's preferred to have the experiment show up in the config by default then we can go with either:
1. A struct with an omitempty bool in it (would show up as `{}` in the config)
2. A flag (would show up as `null` in the config)